client/src/components/VideoForm.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import validUrl from "valid-url";
+
+function VideoForm({ videoData, setVideoData, setFetchData }) {
+  const [errorMessage, setErrorMessage] = useState("");
+
+  function addVideoHandler(event) {
+    event.preventDefault();
+    function validateUrl(urlObject) {
+      return (
+        validUrl.isUri(urlObject) &&
+        (urlObject.startsWith("https://www.youtube.com") ||
+          urlObject.startsWith("https://youtu.be") ||
+          urlObject.startsWith("https://m.youtube.com") ||
+          urlObject.startsWith("https://youtube.com/"))
+      );
+    }
+
+    const formTitle = event.target.form.title.value;
+    const formUrl = event.target.form.url.value;
+
+    const cleanedUrl = formUrl.includes("&") ? formUrl.split("&")[0] : formUrl;
+
+    if (validateUrl(formUrl) && formTitle !== "") {
+      const newData = {
+        id: parseInt(Math.random() * 1000),
+        title: formTitle,
+        url: cleanedUrl,
+        rating: 0,
+        timeSent: Date(),
+      };
+
+      fetch("http://localhost:5000/videos", {
+        method: "post",
+        mode: "cors",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newData),
+      });
+
+      setFetchData(true);
+      setErrorMessage("");
+    } else if (formTitle === "" && !validateUrl(formUrl)) {
+      setErrorMessage("Add a title and a valid URL");
+    } else if (formTitle === "") {
+      setErrorMessage("Add a title");
+    } else {
+      setErrorMessage("Add a valid URL");
```

Comment:
It may be worth describing what a valid URL looks like (e.g. if I put in https://github.com, that's valid URL - you have stricter criteria than that, but you're not telling me what they are :))

Reply:
I have changed my error message to, "Add a valid YouTube video URL". :))